django/thunderstore/account/models/user_flag.py

```diff
@@ -0,0 +1,92 @@
+import datetime
+from typing import List
+
+from django.conf import settings
+from django.contrib.contenttypes.fields import GenericForeignKey
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+from django.db.models import Manager, Q
+
+from thunderstore.core.mixins import TimestampMixin
+from thunderstore.core.types import UserType
+
+
+class UserFlag(TimestampMixin):
+    objects: Manager["UserFlag"]
+
+    name = models.TextField()
+    description = models.TextField(blank=True)
+    app_label = models.CharField(
+        max_length=100,  # Copied from Django's ContentType model
+        help_text="The Django app label of the app that provided this flag type",
+    )
+    identifier = models.SlugField(
+        unique=True,
+        help_text="The globally unique identifier for this flag type",
+    )
+
+    def __str__(self):
+        return f"{self.app_label}: {self.name}"
+
+    @classmethod
+    def get_active_flags_on_user(
+        cls, user: UserType, timestamp: datetime.datetime
```

Comment:
We do need a timestamp for the query yes, but it could have a default value if we wanted to. I figured it's better to be explicit over implicit, and it also leads to less complex code to keep it required (for now at least, since one less if condition), thus why I went with this solution.